### PR TITLE
add knex config, migrations, and seeds; document in readme

### PR DESCRIPTION
**Pull request by ashleygwilliams**
review plz.
solves #23

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
this is ready for review! :bow:

i may branch from here to do the bookshelf models integration

File: migrations/1_create_users_projects_files.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+exports.up = function (knex) {
+  return knex.schema
+    .createTable('users', function (t) {
+      t.increments('id');
+      t.text('name').notNullable().unique();
+    })
+    .createTable('projects', function (t) {
+      t.increments('id');
+      t.integer('user_id').notNullable().references('id').inTable('users');
+      t.text('title').notNullable();
```
Comment (sedge) [inline migrations/1_create_users_projects_files.js]: Aren't we missing files here?

File: migrations/1_create_users_projects_files.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    })
+    .createTable('files', function (t) {
+      t.increments('id');
+      t.integer('project_id').notNullable().references('id').inTable('projects');
+      t.text('path').notNullable();
```
Comment (sedge) [inline migrations/1_create_users_projects_files.js]: Similarly, aren't we missing the data?

**Comment by sedge** [comment]
This looks clean and fantastic. knex++! I left a few questions in the source.

File: README.md [hunk 3]
```diff
@@ -31,7 +31,14 @@ OR, if you are on Windows
 $ COPY env.dist .env
 ```
 
-4) Run the server at the default log level (`'info'`):
+4) Set up your database (currently sqlite3 using knex)
+
+```
+$ knex migrate:latest
```
Comment (gideonthomas) [inline README.md]: this need not be in this patch but maybe we can have an npm script to runt both commands?

**Comment by sedge** [comment]
r+

File: README.md (hunk 3 above)
Comment (ashleygwilliams) [inline README.md]: happy to do it in this patch. commit incoming.

**Comment by gideonthomas** [comment]
this looks great...I think it's ready to squash and land!

**Comment by ashleygwilliams** [comment]
squashed! :fist:

**Comment by ashleygwilliams** [comment]
:dancers:

**Pull request merged**